distributed/deploy/cluster.py

```diff
@@ -50,8 +51,9 @@ class Cluster:
     """
 
     _supports_scaling = True
+    name = str(uuid.uuid4())[:8]
```

Comment:
So each instance of an unnamed cluster gets the same name - why is that useful? This is then used for equality, but I would have thought it takes much more than that for two clusters to be considered equal; I *do* agree on equality in the case that the name is meaningful.

Wouldn't it be better to leave it as `None` when the name is not meaningful, and use `hash()` for equality in this case?

Reply:
> So each instance of an unnamed cluster gets the same name

Each instance of an unnamed cluster is given a unique name. The idea is that you can attach to an existing cluster by name. 

It may help to consider a cluster as more than the Python class which represents it. In many cases it is a collection of resources created in some other infrastructure (pods on kubernetes for example). Giving all of these things a common cluster name allows you to implement discovery downstream where we look at all the bits of an existing cluster, then construct a Python object to represent those.

Reply:
> Each instance of an unnamed cluster is given a unique name.

Shouldn't this line be in __init__ in that case? 

Adding the following fails the test
```diff
--- a/distributed/deploy/tests/test_local.py
+++ b/distributed/deploy/tests/test_local.py
@@ -1059,3 +1059,5 @@ async def test_cluster_names():
             assert unnamed_cluster == unnamed_cluster
             assert named_cluster == named_cluster
             assert unnamed_cluster != named_cluster
+        async with LocalCluster(processes=False, asynchronous=True) as unnamed_cluster2:
+            assert unnamed_cluster2 != unnamed_cluster
```

Reply:
Ah good catch thanks. I've updated the test and code.